src/modeler/AutoRest.Swagger/Validation/ListByOperationsValidation.cs

```diff
@@ -0,0 +1,143 @@
+        private readonly string ListTemplate = "{0}_List";
+        private readonly string ResourceGroup = "ResourceGroup";
+        private readonly string SubscriptionId = "SubscriptionId";
+
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string,Dictionary<string, Operation>> entity, RuleContext context)
+        {
+            // get all operation objects that are either of get or post type
+            var serviceDefinition = (ServiceDefinition)context.Root;
+            
+            foreach (var pathObj in entity)
+            {
+                var listOpIds = pathObj.Value.Select(pair => {
+                    // if the operation is not a get or a post, skip
+                    if (!(pair.Key.Equals("get") || pair.Key.Equals("post")))
+                    { return null; }
+                    // if the operation id is not of the form *_list(by), skip
+                    if (!(ListByRegex.IsMatch(pair.Value.OperationId) || pair.Value.OperationId.ToLower().EndsWith("_list")))
+                    { return null; }
+                    // if the operation does not return an array type or is not an xms pageable type, skip
+                    if (!(AutoRest.Swagger.Model.Utilities.ValidationUtilities.IsXmsPageableOrArrayResponseOperation(pair.Value, serviceDefinition)))
+                    { return null; }
+                    // if all conditions pass, return the operation id
+                    return pair.Value.OperationId;
+                }).Where(opId=> opId!=null);
+
+                // if there are no operations matching our conditions, skip
+                if (IsNullOrEmpty(listOpIds))
+                { continue;  }
+
+                // populate valid list operation names for given path
+                var opNames = GetValidListOpNames(pathObj.Key);
+                
+                // if url does not match any of the predefined regexes, skip
+                if (IsNullOrEmpty(opNames))
+                { continue; }
+
+                // find if there are any operations that violate the rule
+                var errOpIds = listOpIds.Where(opId => !opNames.Contains(opId.ToLower()));
+                // no violations found, skip
+                if (IsNullOrEmpty(errOpIds))
+                { continue; }
+
+                foreach (var errOpId in errOpIds)
+                {
+                    yield return new ValidationMessage(context.Path, this, errOpId);
+                }
+            }
+        }
+
+        private IEnumerable<string> GetValidListOpNames(string path)
+        {
+            var opList = new List<string>();
+            var methodGrp = GetMethodGroup(path);
+            if (methodGrp == string.Empty)
+            { return opList; }
+
+            // construct _listbyparent
+            var match = ParentRegex.Match(path.Substring(0, path.IndexOf(methodGrp)-1));
+            if (match.Success)
+            {
+                var caps = match.Groups[1].Captures;
```

Comment:
think all this can be simplified to
```c#
var cap = match.Groups[1].Captures.Reverse().FirstOrDefault(cap => PathParameterRegex.IsMatch(cap));
if (cap != null)
{
    opList.Add(string.Format(ListByTemplate, methodGrp, cap.Value.Substring(1)).Replace(".", string.Empty).ToLower());
}
```